### PR TITLE
feat(reporter): add check coverage thresholds

### LINKED ISSUE #21: Adding a 'checkCoverage' option to compare coverage with a defined threshold
At the moment it's not possible to run the 'check-coverage' command available in Istanbul. It would be cool to have a way of specifying thresholds for statement coverage, function coverage and branch coverage.

I was thinking something like 

```
checkCoverage: {
   statement : 90,
   branch: 80,
   function: 100
}
```

to be added to the Karma conf file.

Thoughts?

### PR DESCRIPTION
**Pull request by nmalaguti**
Add check option to coverageReporter options. Supports similar
options to istanbul's check-coverage feature.

It will cause karma to return a non-zero exit code if coverage
thresholds are not met.

Closes #21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,6 +193,45 @@ coverageReporter: {
...
+
+For example, `statements: 90` implies minimum statement coverage is 90%. `statements: -10` implies that no more than 10 uncovered statements are allowed.
+
+`global` applies to all files together and `each` on a per-file basis. A list of files or patterns can be excluded from enforcement via the `exclude` property. On a per-file or pattern basis, per-file thresholds can be overridden via the `overrides` property.
+
+```javascript
+coverageReporter: {
+  check: {
+    global: {
+      statements: 50,
```
Comment (stramel) [inline README.md]: What is this threshold? Red to Yellow, or Yellow or Green?

Comment (nmalaguti) [inline README.md]: The check thresholds are separate from the watermark thesholds, so the colors have no impact on whether the check will pass or fail.

The Istanbul tool also keeps them separate.

**Comment by nmalaguti** [comment]
Please review and merge this. Thanks.

**Comment by nmalaguti** [comment]
What is the process for getting this reviewed?

**Comment by stramel** [comment]
@nmalaguti From my experience, waiting or pestering them until they do it. I had a PR that sat out here for months.

**Comment by maksimr** [comment]
@nmalaguti , @stramel sorry guys. I really should spend some time to review because I dont use coverage(karma-coverage) and It's hard understand on the fly what this code do

Thanks!

**Comment by nmalaguti** [comment]
Let me know if there's anything I can do to add some context to the change.

**Comment by nmalaguti** [comment]
Following up on this.

**Comment by nmalaguti** [comment]
@maksimr I wanted to follow up on this and see if we can get this merged in and released.

**Comment by stramel** [comment]
@nmalaguti Way to be persistent :+1:

**Comment by Dignifiedquire** [comment]
I haven't gotten to karma-coverage yet, but will try to get to it this week, if @maksimr doesn't get to it first

**Comment by nmalaguti** [comment]
@Dignifiedquire @maksimr any chance this can be merged this week?

**Comment by nmalaguti** [comment]
@Dignifiedquire @maksimr following up on this PR.

File: lib/reporter.js [hunk 2]
```diff
@@ -67,6 +80,132 @@ var CoverageReporter = function(rootConfig, helper, logger) {
...
+      }
+    });
+
+    return thresholds;
+  }
+
+  /*
+   Copyright (c) 2012, Yahoo! Inc.  All rights reserved.
+   Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
+   */
```
Comment (Dignifiedquire) [inline lib/reporter.js]: Why are you adding these license blocks?

**Comment by nmalaguti** [comment]
The code to calculate coverage was sourced from https://github.com/gotwarlost/istanbul/blob/master/lib/command/check-coverage.js and modified to work in karma. I'm giving proper attribution to the code.

**Comment by Dignifiedquire** [comment]
I'm all for giving attribution, but it's not really helpful to have these multiple license blocks, with references to a license file that doesn't even exist in this code base. I suggest you add a header to the file saying something like

```js
// Parts of this code is based on [1], which is licensed under the new BSD license
// For more information see the license accompanying [1]
// 
// [1]: https://github.com/gotwarlost/istanbul/blob/master/lib/command/check-coverage.js
```

**Comment by Dignifiedquire** [comment]
Also need another rebase after I merged some PRs

**Comment by nmalaguti** [comment]
Rebased. I added the BSD license to the LICENSE file so that it will be distributed with the code. I also added the suggested comment to the top.

**Comment by Dignifiedquire** [comment]
Please put the license in a different file, something like `LICENSE-istanbul`, to make the references clearer.

**Comment by Dignifiedquire** [comment]
That change and one more rebase and I can push it out for a new release :)

**Comment by nmalaguti** [comment]
Rebased and moved LICENSE.

**Comment by Dignifiedquire** [comment]
Thanks for all the work!

**Pull request merged**